Mathlib/Data/Complex/Basic.lean

```diff
@@ -0,0 +1,1376 @@
+#align set.re_prod_im Complex.Set.reProdIm
+
+-- mathport name: «expr ×ℂ »
+infixl:72 " ×ℂ " => Set.reProdIm
+
+theorem mem_reProdIm {z : ℂ} {s t : Set ℝ} : z ∈ s ×ℂ t ↔ z.re ∈ s ∧ z.im ∈ t :=
+  Iff.rfl
+#align complex.mem_re_prod_im Complex.mem_reProdIm
+
+instance : Zero ℂ :=
+  ⟨(0 : ℝ)⟩
+
+instance : Inhabited ℂ :=
+  ⟨0⟩
+
+@[simp]
+theorem zero_re : (0 : ℂ).re = 0 :=
+  rfl
+#align complex.zero_re Complex.zero_re
+
+@[simp]
+theorem zero_im : (0 : ℂ).im = 0 :=
+  rfl
+#align complex.zero_im Complex.zero_im
+
+@[simp, norm_cast]
+theorem ofReal_zero : ((0 : ℝ) : ℂ) = 0 :=
+  rfl
+#align complex.of_real_zero Complex.ofReal_zero
+
+@[simp]
+theorem ofReal_eq_zero {z : ℝ} : (z : ℂ) = 0 ↔ z = 0 :=
+  ofReal_inj
+#align complex.of_real_eq_zero Complex.ofReal_eq_zero
+
+theorem ofReal_ne_zero {z : ℝ} : (z : ℂ) ≠ 0 ↔ z ≠ 0 :=
+  not_congr ofReal_eq_zero
+#align complex.of_real_ne_zero Complex.ofReal_ne_zero
+
+instance : One ℂ :=
+  ⟨(1 : ℝ)⟩
+
+@[simp]
+theorem one_re : (1 : ℂ).re = 1 :=
+  rfl
+#align complex.one_re Complex.one_re
+
+@[simp]
+theorem one_im : (1 : ℂ).im = 0 :=
+  rfl
+#align complex.one_im Complex.one_im
+
+@[simp, norm_cast]
+theorem ofReal_one : ((1 : ℝ) : ℂ) = 1 :=
+  rfl
+#align complex.of_real_one Complex.ofReal_one
+
+@[simp]
+theorem ofReal_eq_one {z : ℝ} : (z : ℂ) = 1 ↔ z = 1 :=
+  ofReal_inj
+#align complex.of_real_eq_one Complex.ofReal_eq_one
+
+theorem ofReal_ne_one {z : ℝ} : (z : ℂ) ≠ 1 ↔ z ≠ 1 :=
+  not_congr ofReal_eq_one
+#align complex.of_real_ne_one Complex.ofReal_ne_one
+
+instance : Add ℂ :=
+  ⟨fun z w => ⟨z.re + w.re, z.im + w.im⟩⟩
+
+@[simp]
+theorem add_re (z w : ℂ) : (z + w).re = z.re + w.re :=
+  rfl
+#align complex.add_re Complex.add_re
+
+@[simp]
+theorem add_im (z w : ℂ) : (z + w).im = z.im + w.im :=
+  rfl
+#align complex.add_im Complex.add_im
+
+section
+set_option linter.deprecated false
+@[simp]
+theorem bit0_re (z : ℂ) : (bit0 z).re = bit0 z.re :=
+  rfl
+#align complex.bit0_re Complex.bit0_re
+
+@[simp]
+theorem bit1_re (z : ℂ) : (bit1 z).re = bit1 z.re :=
+  rfl
+#align complex.bit1_re Complex.bit1_re
+
+@[simp]
+theorem bit0_im (z : ℂ) : (bit0 z).im = bit0 z.im :=
+  Eq.refl _
+#align complex.bit0_im Complex.bit0_im
+
+@[simp]
+theorem bit1_im (z : ℂ) : (bit1 z).im = bit0 z.im :=
+  add_zero _
+#align complex.bit1_im Complex.bit1_im
+
+@[simp, norm_cast]
+theorem ofReal_add (r s : ℝ) : ((r + s : ℝ) : ℂ) = r + s :=
+  ext_iff.2 <| by simp [ofReal']
+#align complex.of_real_add Complex.ofReal_add
+
+-- Porting note: removed norm_cast attribute "rhs can't start with coe"
+-- Was warned that this is a syntactic tautology.
+@[simp, nolint synTaut]
+theorem ofReal_bit0 (r : ℝ) : ((bit0 r : ℝ) : ℂ) = bit0 r  :=
+  ext_iff.2 <| by simp [bit0]
+#align complex.of_real_bit0 Complex.ofReal_bit0
+
+-- Porting note: removed norm_cast attribute "rhs can't start with coe"
+-- Was warned that this is a syntactic tautology.
+@[simp,  nolint synTaut]
+theorem ofReal_bit1 (r : ℝ) : ((bit1 r : ℝ) : ℂ) = bit1 r :=
```

Comment:
same here, and note that `norm_cast` doesn't complain because the RHS doesn't start with a `↑` anymore (as above).
```suggestion
@[simp, norm_cast]
theorem ofReal_bit1 (r : ℝ) : ((bit1 r : ℝ) : ℂ) = bit1 (r : ℂ) :=
```

Reply:
Done